pkg/kubelet/container_bridge.go

```diff
@@ -27,8 +28,39 @@ import (
 
 var cidrRegexp = regexp.MustCompile(`inet ([0-9a-fA-F.:]*/[0-9]*)`)
 
+func createCBR0(wantCIDR *net.IPNet) error {
+	// recreate cbr0 with wantCIDR
+	if err := exec.Command("brctl", "addbr", "cbr0").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	if err := exec.Command("ip", "addr", "add", wantCIDR.String(), "dev", "cbr0").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	if err := exec.Command("ip", "link", "set", "dev", "cbr0", "up").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	// restart docker
+	if err := exec.Command("service", "docker", "restart").Run(); err != nil {
```

Comment:
This assumes a non-systemd distro. Ideally we should support `systemctl restart docker` here which should cover the majority of the systemd only distros such as CoreOS. I see a couple of options here. Allow the user to specify a `--docker-restart-cmd` flag or test for the presence of the `service` command in PATH, and fallback to `systemctl`.